baybe/surrogates/base.py

```diff
@@ -74,45 +96,33 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
-    _input_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
-        init=False, default=None, eq=False
-    )
-    """Callable preparing surrogate inputs for training/prediction.
-
-    Transforms a dataframe containing parameter configurations in experimental
-    representation to a corresponding dataframe containing their **scaled**
-    computational representation. Only available after the surrogate has been fitted."""
+    _searchspace: SearchSpace | None = field(init=False, default=None, eq=False)
+    """The search space on which the surrogate operates. Available after fitting."""
 
-    _output_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
-        init=False, default=None, eq=False
-    )
-    """Callable preparing surrogate outputs for training.
-
-    Transforms a dataframe containing target measurements in experimental
-    representation to a corresponding dataframe containing their **scaled**
-    computational representation. Only available after the surrogate has been fitted."""
-
-    # TODO: type should be `Standardize | _NoTransform`` but is currently
-    #   omitted due to: https://github.com/python-attrs/cattrs/issues/531
+    # TODO: type should be
+    #   `botorch.models.transforms.outcome.Standardize | _NoTransform`
+    #   but is currently omitted due to:
+    #   https://github.com/python-attrs/cattrs/issues/531
     _output_scaler = field(init=False, default=None, eq=False)
-    """Optional callable for scaling output values.
+    """Scaler for transforming output values. Available after fitting.
 
     Scales a tensor containing target measurements in computational representation
-    to make them ready for processing by the surrogate. Only available after the
-    surrogate has been fitted."""
+    to make them digestible for the model-specific, scale-agnostic posterior logic."""
 
-    def to_botorch(self) -> Model:
-        """Create the botorch-ready representation of the model."""
+    def to_botorch(self) -> Model:  # noqa: D102
+        # See base class.
         from baybe.surrogates._adapter import AdapterModel
 
         return AdapterModel(self)
 
     @staticmethod
-    def _make_parameter_scaler(
+    def _make_parameter_scaler_factory(
         parameter: Parameter,
-    ) -> ParameterScalerProtocol | None:
-        """Return the scaler to be used for the given parameter."""
-        return MinMaxScaler()
+    ) -> type[InputTransform] | None:
+        """Return the scaler factory to be used for the given parameter."""
+        from botorch.models.transforms.input import Normalize
+
+        return Normalize
 
     @staticmethod
     def _make_target_scaler() -> OutcomeTransform | None:
```

Comment:
Why do we have `None` here but `_NoTransform` in other places? This seems a bit unconsistent to me.

Reply:
The reason is that `_NoTransform` is required only for method whose return value is stored in the surrogate attributes, to avoid ambiguity with `None`:
* `None` is the default value of the attributes, indicating that the surrogate has not yet been fitted
* If I also used `None` in the special case where no scaling is used, then the state of the attribute would be ambiguous since no longer clear if unfitted or no transform desired. Hence, I use `_NoTransform` as a sentinel value.

We *could* also use it in the lower-level methods `_make_target_scaler` and `_make_parameter_scaler_factory` but that would only be for readability, not changing anything from a functional point of view. 
@Scienfitz: what do you think?

Reply:
the need for the sentinel of output scaler seems clear to me
I would not prefer to have that also for target/parameter scalers simply because it would mean users (who are expected to overwrite target/param scalers but no input/output scalers) would need the knowledge of this sentinel etc. It is much simpler for them if they can simply return None if they want no scaling for the respective target/parameter

Reply:
Alright, will resolve then